spotify/src/connector.js

```diff
@@ -0,0 +1,129 @@
+var spotifySchema = [
+  {
+    name: 'text',
+    dataType: 'STRING',
+    semantics: {
+      conceptType: 'DIMENSION'
+    }
+  },
+  {
+    name: 'popularity',
+    dataType: 'NUMBER',
+    semantics: {
+      conceptType: 'METRIC'
+    }
+  }
+];
+function getAuthType() {
+  var response = {
+    "type": "OAUTH2"
+  };
+  return response;
+}
+var getOAuthService = function() {
+  return OAuth2.createService('Spotify')
+    .setAuthorizationBaseUrl('https://accounts.spotify.com/authorize?response_type=code')
+    .setTokenUrl('https://accounts.spotify.com/api/token')
+    .setClientId('b475f97956fe4b6bb66cfe0a239254a9')
```

Comment:
The clientId and clientSecret should be put in the PropertiesService. I'd recommend revoking this clientSecret, and adding a new one via PropertiesService.